js/modules/k6/experimental/streams/tests/testharness.js.patch

```diff
@@ -0,0 +1,46 @@
+diff --git a/resources/testharness.js b/resources/testharness.js
+index c5c375e17..57d201b57 100644
+--- a/resources/testharness.js
++++ b/resources/testharness.js
+@@ -487,7 +487,11 @@
+         this.all_loaded = false;
+         this.on_loaded_callback = null;
+         Promise.resolve().then(function() {
+-            this.all_loaded = true
++            // This is the test environment used when we run tests from Go
++            // code and with the Goja/k6 runtime. However, we don't want
++            // all tests to be marked as loaded "by default", as it would
++            // make some of the tests to not be executed at all.
++            // this.all_loaded = true
+             if (this.on_loaded_callback) {
+                 this.on_loaded_callback();
+             }
+@@ -2631,6 +2635,16 @@
+             this.set_status(status, message, stack);
+             this.phase = this.phases.HAS_RESULT;
+             this.done();
++
++						// We don't want to rely on the DOM and other browser-based
++            // mechanisms for reporting test failures. Instead, we just
++            // throw the error and make it fail fast, to be aware of it.
++            //
++            // In the future, we might want to add a way to report these
++            // that doesn't rely on a browser, but let all test end before
++            // actually reporting failures.
++            // But, as a first iteration, this approach should suffice.
++            throw `${this.name} failed - ${e}`;
```

Comment:
I guess here `set_status` above should be the thing that marks tests as failed?

Reply:
Yeah, but that status isn't directly reflected in any way through how we do run the tests. That's why I suggested there to explore a mechanism to do so, like "building" our own test environment, but I'd prefer to leave that for the next iteration as for now it's more than enough (it reports the failures) and the major drawback is that it fail fast and doesn't run all the tests when there's an error, which shouldn't be that bad for an initial release, as long as the behavior is correct.

(cc/ @oleiade as you'll most likely take over 🙇🏻)